core/tools/blueos_startup_update/blueos_startup_update

```diff
@@ -75,18 +75,54 @@ def update_cgroups() -> None:
 
 def update_dwc2() -> None:
     print("Running dwc2 update..")
-    # Enable USB OTG as ethernet adapter
-    command = '''sudo sh -c '\
-    grep -q modules-load=dwc2 /boot/cmdline.txt || (
-        truncate --size -1 /boot/cmdline.txt;\
-        echo " modules-load=dwc2,g_ether" >> /boot/cmdline.txt;\
-    )' '''
+
+    # First, remove any dwc2 overlay configuration
+    config_file = '/boot/config.txt'
+    dwc2_overlay_match_pattern = '/^dtoverlay=dwc2.*$/d'
+    command = f'''\
+    sudo sed --in-place "{dwc2_overlay_match_pattern}" "{config_file}"
+    '''
     print(run_command(command, False))
 
-    command = '''sudo sh -c '\
-    grep -q "dtoverlay=dwc2" /boot/config.txt || (
-        echo "dtoverlay=dwc2" | sudo tee -a /boot/config.txt
-    )' '''
+    # Create a pi4 section at the end if it doesn't exist
+    cm4_section_match_pattern = '^\\[pi4\\].*$'
```

Comment:
why "cm4" ?

Reply:
> what if instead of doing a lot of seds and greps, we load the file using cat and commander and then put it back there?
> 
> Then maybe we could have more legible code. there are probably libraries we can leverage on to parse/edit the file in a better way, too.

I agree, I tried the parserconfig but this config file is incompatible because it doesn't present a header... Which if loading with `cat` we could add and later filter out... But before doing this workaround I'd take a look on other options :)

Reply:
> why "cm4" ?
Lol, should be `pi4`